kubespawner/spawner.py

```diff
@@ -583,15 +583,30 @@ def _hub_connect_port_default(self):
     )
 
     def _expand_user_properties(self, template):
-        # Make sure username matches the restrictions for DNS labels
+        # Make sure username and servername match the restrictions for DNS labels
         safe_chars = set(string.ascii_lowercase + string.digits)
-        legacy_escaped_username = ''.join([s if s in safe_chars else '-' for s in self.user.name.lower()])
-        safe_username = escapism.escape(self.user.name, safe=safe_chars, escape_char='-').lower()
-        return template.format(
-            userid=self.user.id,
-            username=safe_username,
-            legacy_escape_username=legacy_escaped_username
-        )
+        # Test if a named-server exists
+        if getattr(self, 'name', None) is None:
+            legacy_escaped_username = ''.join([s if s in safe_chars else '-' for s in self.user.name.lower()])
+            safe_username = escapism.escape(self.user.name, safe=safe_chars, escape_char='-').lower()
+            return template.format(
+                userid=self.user.id,
+                username=safe_username,
+                legacy_escape_username=legacy_escaped_username
+                )
+        # If a named-server exists, then extend the properties    
+        else:
+            legacy_escaped_username = ''.join([s if s in safe_chars else '-' for s in self.user.name.lower()])
+            safe_username = escapism.escape(self.user.name, safe=safe_chars, escape_char='-').lower()
+            legacy_escaped_servername = ''.join([s if s in safe_chars else '-' for s in self.name.lower()])
+            safe_servername = escapism.escape(self.name, safe=safe_chars, escape_char='-').lower()
+            return template.format(
+                userid=self.user.id,
+                username=safe_username,
+                legacy_escape_username=legacy_escaped_username,
+                servername=safe_servername,
+                legacy_escape_servername=legacy_escaped_servername
```

Comment:
Probably don't need `legacy_escape_servername` since this wasn't present before.